cmd/dlt/oidcprovider/cmd.go

```diff
@@ -78,16 +78,19 @@ func run(cmd *cobra.Command, argv []string) {
 		os.Exit(1)
 	}
 
-	clusterID := clusterKey
 	if sub != nil {
-		clusterID = sub.ClusterID()
+		clusterKey = sub.ClusterID()
 	}
-	c, err := r.OCMClient.GetClusterByID(clusterID, r.Creator)
+	c, err := r.OCMClient.GetCluster(clusterKey, r.Creator)
 	if err != nil {
 		if errors.GetType(err) != errors.NotFound {
 			r.Reporter.Errorf("Error validating cluster '%s': %v", clusterKey, err)
 			os.Exit(1)
+		} else if sub == nil {
```

Comment:
Hey there sorry for taking a while to get to this and commenting.

Not sure if I follow this check, could you provide some more info in why this is needed?

Reply:
Hey @gdbranco , no worries, figured end of year is a pretty busy time for all.

The `if sub == nil` check here is to identify if the cluster ever existed at all.  `GetCluster` and `GetClusterByID` will return `errors.NotFound` for a cluster that has been deleted. The assumption is that `errors.NotFound` means the cluster has been deprovisioned and is thus in a state where the OIDC Provider can be deleted, and that `sub` can be used to find the value of the cluster ID.

This is not an accurate assumption for a cluster that never existed at all.  If `GetCluster` cannot find the cluster and `sub == nil`, then the cluster never existed, and we should not attempt to delete an OIDC Provider.

Without the `sub == nil` check, ` r.AWSClient.GetOpenIDConnectProvider(sub.ClusterID())` is called, which results in an empty string being sent, causing the first OIDC provider found by the AWS API to be returned and later deleted.

You can see that behavior by attempting to delete the OIDC provider for a cluster that never existed, in an account where at least one OIDC provider does exist.
```
rosa delete oidc-provider -m manual -c cluster-that-never-existed-at-all
I: Run the following commands to delete the OIDC provider:

aws iam delete-open-id-connect-provider \
	--open-id-connect-provider-arn arn:aws:iam::1111222233334444:oidc-provider/rh-oidc.s3.us-east-1.amazonaws.com/xxxxxxxxxxxxxx
```

Reply:
Would it then make sense to fail early when fetching the sub? I mean in line 70 above:
```
sub, err := r.OCMClient.GetClusterUsingSubscription(clusterKey, r.Creator)
if sub == nil {
  r.Reporter.Errorf("Failed to get cluster '%s': Subscription not found", r.ClusterKey)
  os.Exit(1)
}
```

Reply:
@vkareh Did my explanation make sense?  I think as the PR stands currently, it's the least amount of code, touching only the parts of the logic that are currently broken.

If you'd rather take a step back and look at a broader change to the logic to make it more clear, let me know and I can work on that too.  Even just refactoring `GetClusterUsingSubscription` to `GetDeprovisionedClusterUsingSubscription` would make it a bit more clear.

Reply:
@jjulien yes, this makes perfect sense. Thanks for the clarification!